src/caldav-calendar.c

```diff
@@ -505,11 +505,13 @@ static void sync_multiget_report_done(CURL* curl, CURLcode ret, void* user)
 		// but its caldata member will be empty. So it can be ignored.
 		if (cde->caldata) {
 			Event* event = create_event_from_parsed_xml(rc, cde);
-			sc->cal->events = g_slist_append(sc->cal->events, event);
-			nNew++;
-		} else {
-			caldav_entry_free(cde);
+			if (event) {
+				sc->cal->events = g_slist_append(sc->cal->events, event);
+				nNew++;
+				continue;
+			}
 		}
+		caldav_entry_free(cde);
```

Comment:
Maybe it's just personal preference, but I think this can be horizontally squished a bit and made a little clearer without `continue` (I'm totally aware that I used `goto` in this file though...)

I'll push a little tweak